### PR TITLE
Add optimized building tools

### LINKED ISSUE #58: Research about the generation of OS packages (rpm, deb)
## Description

We need to investigate how to produce production packages in rpm and deb formats.

Currently, we only generate production packages in `tar.gz` format. See #46.

## Additional info
These are the tools used by CI/CD:

- https://github.com/wazuh/wazuh-packages/tree/master/stack/indexer/rpm
- https://github.com/wazuh/wazuh-packages/tree/master/stack/indexer/deb
- https://github.com/wazuh/wazuh-packages/tree/master/stack/dashboard/rpm
- https://github.com/wazuh/wazuh-packages/tree/master/stack/dashboard/deb

### PR DESCRIPTION
**Pull request by Tostti**
### Description
This PR aims to add the necessary tools to build `rpm` and `deb` packages in an efficient way.
 
### Issues Resolved
#58 
 
### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [x] `yarn test:ftr`
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: dev-tools/build-packages/deb/debian/compat [hunk 1]
```diff
@@ -0,0 +1 @@
+8
```
Comment (AlexRuiz7) [inline dev-tools/build-packages/deb/debian/compat:1]: Getting this output
```
dh: Compatibility levels before 9 are deprecated (level 8 in use)
```

File: dev-tools/build-packages/deb/debian/rules [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	cp $(TARGET_DIR)$(INSTALLATION_DIR)/wazuh-dashboard.service $(TARGET_DIR)/etc/systemd/system/wazuh-dashboard.service
+	cp $(TARGET_DIR)$(INSTALLATION_DIR)/wazuh-dashboard $(TARGET_DIR)/etc/systemd/system/$(NAME)
+	cp $(TARGET_DIR)$(INSTALLATION_DIR)/default $(TARGET_DIR)/etc/default/$(NAME)
+
+	rm -rf $(TARGET_DIR)$(INSTALLATION_DIR)/etc
+
+	chown -R $(USER):$(GROUP) $(TARGET_DIR)$(INSTALLATION_DIR)
+	chown -R $(USER):$(GROUP) $(TARGET_DIR)$(CONFIG_DIR)
+
+	find $(TARGET_DIR)$(INSTALLATION_DIR)/plugins/wazuh/ -exec chown $(USER):$(GROUP) {} \;
```
Comment (AlexRuiz7) [inline dev-tools/build-packages/deb/debian/rules:82]:
```
make[1]: Entering directory '/build/wazuh-dashboard/wazuh-dashboard-4.6.0'
cp /opt/wazuh-dashboard.tar.gz ./
cp: -r not specified; omitting directory '/opt/wazuh-dashboard.tar.gz'
make[1]: *** [debian/rules:55: override_dh_install] Error 1
make[1]: Leaving directory '/build/wazuh-dashboard/wazuh-dashboard-4.6.0'
make: *** [debian/rules:38: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -b failed
```

Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/debian/rules:82]: Command used:

```
bash deb/launcher.sh \
    --package file://$out_dir/wazuh-dashboard-4.6.0-2-linux-x64.tar.gz \
    --version 4.6.0  \
    --revision 2     \
    --store $out_dir
```

Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/debian/rules:82]: Worked with the same command after changing debian/compat to 9

File: dev-tools/build-packages/deb/debian/compat (hunk 1 above)
Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/debian/compat:1]: Level 9 works.

Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/debian/compat:1]: Level 10 works.

Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/debian/compat]: Changed to `debhelper-compat (= 12)` in `control` file.

`compat` file delete as it was no longer needed.

Source:
- [Guide for Debian Maintainers - Section 5.19.1](https://www.debian.org/doc/manuals/debmake-doc/debmake-doc.en.pdf)
- [OpenSearch build - 2.8.0](https://github.com/opensearch-project/opensearch-build/tree/opensearch-2.8.0/scripts/pkg/build_templates/opensearch-dashboards/deb/debian) - no debian/compat file here

File: dev-tools/build-packages/deb/builder.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# Installing build dependencies
+cd ${source_dir}
+mk-build-deps -ir -t "apt-get -o Debug::pkgProblemResolver=yes -y"
+
+# Build package
+debuild --no-lintian -b -uc -us \
+    -eINSTALLATION_DIR="${directory_base}" \
+    -eVERSION="${version}" \
+    -eREVISION="${revision}" \
+    -eURL="${url}"
```
Comment (AlexRuiz7) [inline dev-tools/build-packages/deb/builder.sh:49]: url is never defined

Comment (AlexRuiz7) [reply dev-tools/build-packages/deb/builder.sh]: Removed in 8cb211e4fda7370008bebffd7ea92e7006c33b0a

**Review by Desvelao — APPROVED** [review]
LGTM!

**Comment by AlexRuiz7** [comment]
Backport to #62 and #68

**Pull request merged**